src/scripts/init_blockchain.sh

```diff
@@ -56,10 +62,17 @@ done
 echo "=== lamington: activate protocol features ==="
 curl --silent --output /dev/null -X POST localhost:8888/v1/producer/schedule_protocol_feature_activations \
   -d '{"protocol_features_to_activate": ["0ec7e080177b2c02b278d5088611686b49d739925a92d9bfcacd7fc6b74053bd"]}'
-sleep 2s
+sleep 1s
 
 echo "=== lamington: install system contracts ==="
+
+cleos set contract eosio "$contracts_dir/eosio.system" -p eosio@active
+sleep 0.5s
+cleos set contract eosio "$contracts_dir/eosio.system" -p eosio@active
+sleep 0.5s
```

Comment:
Does this need to be set 3 times?

Reply:
I'm ashamed of this 😞 but since setting the system contract would often fail without error I found setting it multiple times with a small delay made it work. I really don't like it and would prefer a different option but I was very surprised to see that all my EOSDAC tests had been running without the system contract getting set. It wasn't until I was working on a custom `eosio.msig` without inner deferred transactions that needed `privileged` access that I realised I needed this to be working. I would love a better suggestion because other than just being dirty code it also adds delay to start of tests.

Reply:
Yeah, this feels like something we can ask Block One about, but I've found similar flakiness with EOS setup tasks. Things that should be deterministic just often aren't, then when you ask for how they can automatically spin up a new mainnet with an identical setup they point you to an instruction sheet instead of a script or program. Quite frustrating, definitely not your fault. I'll add a comment noting the reason for this, otherwise it's in a fairly black box part of the system so it shouldn't get in too many people's way.